src/bin/adbackup.rs

```diff
@@ -1,17 +1,41 @@
 #![cfg_attr(feature = "clippy", feature(plugin))]
 #![cfg_attr(feature = "clippy", plugin(clippy))]
 
-#![recursion_limit = "1024"]
+#[macro_use]
+extern crate log;
 
 #[macro_use]
 extern crate clap;
 
-use clap::{App, AppSettings};
+use clap::{Arg, App, AppSettings, SubCommand};
 
 extern crate adbackup;
 
+use adbackup::devices::Device;
+use adbackup::logging;
+
+extern crate failure;
+
+use failure::Error;
+
 fn main() {
-    let _ = make_clap().get_matches();
+    let matches = make_clap().get_matches();
+
+    let (sub_name, subm) = matches.subcommand();
+    let sub_fn = match sub_name {
+        "devices" => print_devices,
+        _ => unimplemented!(),
+    };
+
+    let verbosity = matches.occurrences_of("verbose")
+        + subm.unwrap().occurrences_of("verbose");
```

Comment:
can we get rid of this unwrap or am I missing something and this is something not possible with `failure`?

Reply:
In this case we can't use `failure` because of `subm` is an Option-type...unwrapping in general is a huge mistake, unwrapping an Option is even worse, so thx, will fix that. 
Hacked a bit around to get ride on how clap works but should have refactored the unwrapping-shit before committing^^ maybe we should setup our IDE to display a warning if something unwraps something...